packages/assets-controllers/src/NftDetectionController.test.ts

```diff
@@ -195,669 +127,391 @@ describe('NftDetectionController', () => {
   });
 
   afterEach(() => {
-    nftDetection.stopAllPolling();
     clock.restore();
     sinon.restore();
   });
 
-  it('should set default config', () => {
-    preferences.setUseNftDetection(false);
-    expect(nftDetection.config).toStrictEqual({
-      interval: DEFAULT_INTERVAL,
-      chainId: toHex(1),
-      selectedAddress: '',
-      disabled: true,
+  it('should set default config', async () => {
+    await withController(({ controller }) => {
+      expect(controller.config).toStrictEqual({
+        interval: DEFAULT_INTERVAL,
+        chainId: toHex(1),
+        selectedAddress: '',
+        disabled: true,
+      });
     });
   });
 
   it('should poll and detect NFTs on interval while on mainnet', async () => {
-    const mockNfts = sinon.stub(NftDetectionController.prototype, 'detectNfts');
-    const nftsDetectionController = new NftDetectionController(
-      {
-        getNetworkClientById: jest.fn(),
-        chainId: ChainId.mainnet,
-        onNftsStateChange: (listener) => nftController.subscribe(listener),
-        onPreferencesStateChange: (listener) => preferences.subscribe(listener),
-        onNetworkStateChange: networkStateChangeNoop,
-        getOpenSeaApiKey: () => nftController.openSeaApiKey,
-        addNft: nftController.addNft.bind(nftController),
-        getNftApi: nftController.getNftApi.bind(nftController),
-        getNftState: () => nftController.state,
+    await withController(
+      { config: { interval: 10 } },
+      async ({ controller, triggerPreferencesStateChange }) => {
+        const mockNfts = sinon.stub(controller, 'detectNfts');
+        triggerPreferencesStateChange({
+          ...getDefaultPreferencesState(),
+          useNftDetection: true,
+        });
+        // Wait for detect call triggered by preferences state change to settle
+        await advanceTime({
+          clock,
+          duration: 1,
+        });
+        mockNfts.reset();
+
+        await controller.start();
```

Comment:
Is this necessary? It seems that the controller self-starts when `useNftDetection` is enabled.

Reply:
Hmm. Good point. The way polling is setup right now doesn't really make sense. Not only does it auto-start, it also fails to start at all if the current network isn't Ethereum Mainnet (even if you later switch to Ethereum Mainnet).

For now I'll remove these unnecessary steps, but I'll follow up later with a PR that properly separates the `disabled` state from the `useNftDetection` setting, which will make this API make sense again.

Addressed in ac26f42d